### PR TITLE
Wrldbaems 87

### PR DESCRIPTION
**Pull request by adobe-rpal**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix [WRLDBAEMS-87](https://adobe-acs-group.atlassian.net/browse/WRLDBAEMS-87)

Test URLs:
- Before: https://main--world-bank--aemsites.hlx.live/
- After: [https://WRLDBAEMS-87--world-bank--aemsites.hlx.live/](https://wrldbaems-87--world-bank--aemsites.hlx.live/)

**Authoring URL**: https://author-p136806-e1403562.adobeaemcloud.com/ui#/@worldbank/aem/universal-editor/canvas/author-p136806-e1403562.adobeaemcloud.com/content/world-bank/global/en/drafts/Rajesh/carousel.html

**Test Page URL**: /en/drafts/rajesh/carousel

### CODE REVIEW

**Comment by thomsebastin** [comment]
@adobe-rpal A11y has gone down. Also, I don't see the carousel on the linked page. Kindly check again.

File: styles/carousel.css [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  background-color: #fff;
+  box-shadow: none;
+ }
+
+ .cards-container > .default-content-wrapper {
+  flex: 1;
+  padding: 20px;
+ }
+
+ .cards-container > .default-content-wrapper > div:first-child {
```
Comment (mkbansal1) [inline styles/carousel.css:14]: cards is a different block, which use class name .cards
Please use other class name

File: scripts/carousel.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          moveLeftBtn.style.display = dir === 'ltr' ? 'block' : 'none';
+          moveRightBtn.style.display = dir === 'ltr' ? 'none' : 'block';
+        } else {
+          moveLeftBtn.style.display = 'block';
+          moveRightBtn.style.display = 'block';
+        }
+      } catch (e) {
+        /* error structure was not as expected */
+      }
+    };
```
Comment (thomsebastin) [inline scripts/carousel.js:92]: Should this be inside a try catch?

File: styles/carousel.css [hunk 3]
```diff
@@ -0,0 +1,273 @@
+.cards-container {
+    display: flex;
```
Comment (mrugesh-adobe) [inline styles/carousel.css:2]: Is this a component, if yes it should be within the blocks folder

File: styles/carousel.css [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    padding: 0 72px;
+    background-color: #fff;
+    box-shadow: none;
+  }
+  
+  .cards-container>.default-content-wrapper {
```
Comment (mrugesh-adobe) [inline styles/carousel.css:8]: All root selector should starts with .carousel as it is specific to carousel. I see all root selectors are .cards-container which will collide with cards component

**Review by mrugesh-adobe — CHANGES_REQUESTED** [review]

File: scripts/carousel.js (hunk 2 above)
Comment (adobe-rpal) [reply scripts/carousel.js]: yes, for the conditions in if/else blocks.

File: styles/carousel.css (hunk 3 above)
Comment (adobe-rpal) [reply styles/carousel.css]: Changes done.

File: styles/carousel.css (hunk 4 above)
Comment (adobe-rpal) [reply styles/carousel.css]: Changes done

File: styles/carousel.css (hunk 1 above)
Comment (adobe-rpal) [reply styles/carousel.css]: Done

File: blocks/carousel/carousel.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    if (i > 3) {
+      const li = document.createElement('li');
+      moveInstrumentation(row, li);
+      while (row.firstElementChild) li.append(row.firstElementChild);
+      [...li.children].forEach((div) => {
+        if (div.children.length === 1 && div.querySelector('picture')) div.className = 'cards-card-image';
+        else div.className = 'cards-card-body';
+      });
+      ul.append(li);
+    } else if (i <= 3) {
```
Comment (thomsebastin) [inline blocks/carousel/carousel.js:19]: Just use else here?

File: blocks/carousel/carousel.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { createOptimizedPicture } from '../../scripts/aem.js';
+import { moveInstrumentation } from '../../scripts/scripts.js';
+import createSlider from '../../scripts/slider.js';
+
+export default function decorate(block) {
+  let i = 0;
+  const ul = document.createElement('ul');
+  const divtag = document.createElement('div');
```
Comment (thomsebastin) [inline blocks/carousel/carousel.js:8]: What are these elements in the actual design? Please see if you can add meaningful names.

**Review by thomsebastin — CHANGES_REQUESTED** [review]

File: blocks/carousel/carousel.js (hunk 6 above)
Comment (adobe-rpal) [reply blocks/carousel/carousel.js]: Done

**Review by thomsebastin — APPROVED** [review]

**Comment by mkbansal1** [comment]
@adobe-rpal 

1. Left side text is not aligned properly
EDS:
image

Figma:
image

**Comment by mkbansal1** [comment]
3. Arrows should be on right side
EDS:
image

Figma:
image

**Comment by mkbansal1** [comment]
4. Image corner should be round

**Review by mkbansal1 — CHANGES_REQUESTED** [review]
@adobe-rpal Please see my comments

**Comment by adobe-rpal** [comment]
> 1. is

**Comment by adobe-rpal** [comment]
> @adobe-rpal
> 
> 1. Left side text is not aligned properly
>    EDS:
> 
> image
> Figma: image

- Done

**Comment by adobe-rpal** [comment]
> 2. Remove the margin from bottom text
> 
> image

- Done

**Comment by adobe-rpal** [comment]
> 3. Arrows should be on right side
>    EDS:
> 
> image
> Figma: image

-Done

**Comment by adobe-rpal** [comment]
> 4. Image corner should be round

- Done

**Comment by adobe-rpal** [comment]
> 5. Card - fields should be required
> 
> image

- Done

**Review by mkbansal1 — APPROVED** [review]

**Review by mrugesh-adobe — APPROVED** [review]

**Pull request merged**